proposals/0233-codepoint-and-character-literals.md

```diff
@@ -0,0 +1,234 @@
+# Integer-convertible character literals
+
+* Proposal: SE-0240
+* Authors: [Kelvin Ma](https://github.com/kelvin13) ([@*taylorswift*](https://forums.swift.org/u/taylorswift/summary)), [Chris Lattner](https://github.com/lattner) ([@*Chris_Lattner3*](https://forums.swift.org/u/Chris_Lattner3/summary)), [John Holdsworth](https://github.com/johnno1962) ([@*johnno1962*](https://forums.swift.org/u/johnno1962/summary))
+* Review manager: 
+* Status: *Awaiting review*
+* Implementation (WIP): [`johnno1962:character-literalsII`](https://github.com/johnno1962/swift/tree/character-literalsII)
+* Threads: [1](https://forums.swift.org/t/prepitch-character-integer-literals/10442)
+
+## Introduction
+
+Swift’s `String` type is designed for Unicode correctness and abstracts away the underlying binary representation of the string to model it as a `Collection` of grapheme clusters. This is an appropriate string model for human-readable text, as to a human reader, the atomic unit of a string is (usually) the extended grapheme cluster. When treated this way, many logical string operations “just work” the way users expect. 
+
+However, it is also common in programming to need to express values which are intrinsically numeric, but have textual meaning, when taken as an ASCII value. We propose adding a new literal syntax takes single-quotes (`'`), and is transparently convertible to Swift’s integer types. This syntax, but not the behavior, will extend to all “scalar” text literals, up to and including `Character`, and will become the preferred literal syntax these types.
+
+## Motivation 
+
+For both correctness and efficiency, `[UInt8]` (or another integer array type) is usually the most appropriate representation for an ASCII string. (See [*Stop converting `Data` to `String`*](https://gist.github.com/kelvin13/516a5e3bc699a6b72009ff23f836a4bd) for a discussion on why `String` is an *inappropriate* representation.)
+
+A major pain point of integer arrays is that they lack a clear and readable literal type. In C, `'a'` is a `uint8_t` literal, equivalent to `97`. Swift has no such equivalent, requiring awkward spellings like `UInt8(ascii: "a")`. Alternatives, like spelling out the values in hex or decimal directly, are even worse. This harms readability of code, and is one of the sore points of bytestring processing in Swift.
```

Comment:
Just make this point be about "Things that need the integer ascii or unicode value of a character are hard", the point isn't necessarily about arrays.

Reply:
done